frontend/lib/resolution/disambiguation.cpp

```diff
@@ -1096,6 +1370,255 @@ moreVisible(const DisambiguationContext& dctx,
                           fn1Id, fn2Id);
 }
 
+/**
+  Compare two argument mappings, given a set of actual arguments, and set the
+  disambiguation state appropriately.
+
+  This function implements the argument mapping comparison component of the
+  disambiguation procedure as detailed in section 13.14.3 of the Chapel
+  language specification (page 107).
+
+  actualIdx is the index within the call of the argument to be compared.
+
+  Sets bits in DisambiguationState ds according to whether argument actualIdx
+  in candidate1 vs candidate2 is a better match.
+
+  Returns:
+   -1 if the two formals are incomparable
+    0 if the two formals have the same level of preference
+    1 if fn1 is preferred
+    2 if fn2 is preferred
+ */
+static int testArgMapping(const DisambiguationContext& dctx,
+                          const DisambiguationCandidate& candidate1,
+                          const DisambiguationCandidate& candidate2,
+                          int actualIdx,
+                          DisambiguationState& ds,
+                          std::string& reason) {
+
+  const FormalActual* fa1 = candidate1.formalActualMap.byActualIdx(actualIdx);
+  const FormalActual* fa2 = candidate2.formalActualMap.byActualIdx(actualIdx);
+
+  if (fa1 == nullptr || fa2 == nullptr) {
+
+    return testOpArgMapping(dctx, candidate1, candidate2, actualIdx, ds, reason);
+  }
+
+  QualifiedType f1Type = fa1->formalType();
+  QualifiedType f2Type = fa2->formalType();
+  QualifiedType actualType = fa1->actualType();
+  CHPL_ASSERT(actualType == fa2->actualType());
+
+  // Give up early for out intent arguments
+  // (these don't impact candidate selection)
+  if (f1Type.kind() == QualifiedType::OUT ||
+      f2Type.kind() == QualifiedType::OUT) {
+    return -1;
+  }
+
+  // We only want to deal with the value types here, avoiding odd overloads
+  // working (or not) due to _ref.
+  // TODO: not sure how to reproduce this code in Dyno
+
+  // Additionally, ignore the difference between referential tuples
+  // and value tuples.
+  // TODO: not sure how to reproduce this code in Dyno
+
+  // TODO: Not sure we still need this here, based on production
```

Comment:
Yes I don't know if we need this either. There are some other initializer workarounds in some of your PRs here that we might not need. IMO this comment is sufficient for now.